### PR TITLE
Flake8-annotations: fix errors in openfl-workspace, part 1

### PR DESCRIPTION
**Pull request by ViktoriiaRomanova**
Preparations to add [Flake8-annotations](https://pypi.org/project/flake8-annotations/) plugin (**optional changes**).

This fix is only for errors in **openfl-workspace** folder.

It is assumed that the next flake8-annotations errors will be ignored:

ANN002 - Missing type annotation for `*args`
ANN003 - Missing type annotation for `**kwargs`
ANN101 - Missing type annotation for `self` in method
ANN102 - Missing type annotation for `cls` in classmethod

### CODE REVIEW

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 1]
```diff
@@ -15,7 +18,8 @@
 logger = getLogger(__name__)
 
 
-def download_data_():
+@staticmethod
+def download_data_() -> Any:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]:
```suggestion
def download_data_() -> str:
```

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 2]
```diff
@@ -59,7 +63,8 @@ def download_data_():
         return ''
 
 
-def import_raw_data_(data_path='', num_samples=0):
+def import_raw_data_(data_path: Any = '',
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]:
```suggestion
def import_raw_data_(data_path: str= '',
```

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 3]
```diff
@@ -59,7 +63,8 @@ def download_data_():
...
+                     num_samples: int = 0) -> Tuple[Dict, np.ndarray, np.ndarray, np.ndarray]:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]:
```suggestion
                     num_samples: int = 0) -> Tuple[Dict[str, int], np.ndarray, np.ndarray, np.ndarray]:
```

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 4]
```diff
@@ -146,8 +151,8 @@ def import_raw_data_(data_path='', num_samples=0):
     return details, encoder_input_data, decoder_input_data, decoder_target_data
 
 
-def get_datasets_(encoder_input_data, decoder_input_data,
-                  decoder_target_data, num_samples, split_ratio):
+def get_datasets_(encoder_input_data: np.ndarray, decoder_input_data: np.ndarray,
+                  decoder_target_data: np.ndarray, num_samples: int, split_ratio: float) -> Dict:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]:
```suggestion
                  decoder_target_data: np.ndarray, num_samples: int, split_ratio: float) -> Dict[str, np.ndarray]:
```

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 5]
```diff
@@ -181,7 +186,8 @@ def get_datasets_(encoder_input_data, decoder_input_data,
     return results
 
 
-def load_shard(collaborator_count, shard_num, data_path, num_samples, split_ratio):
+def load_shard(collaborator_count: int, shard_num: Any, data_path: str,
+               num_samples: int, split_ratio: float) -> Tuple:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]: Can we clarify the returned type here?

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 6]
```diff
@@ -47,11 +51,11 @@ def __init__(self, collaborator_count, split_ratio,
         self.X_valid = [valid[0], valid[1]]
         self.y_valid = valid[2]
 
-    def get_feature_shape(self):
+    def get_feature_shape(self) -> tuple:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]: Can we use `Tuple` and clarify type inside it?

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 7]
```diff
@@ -91,7 +95,10 @@ def get_valid_data_size(self):
         return self.X_valid[0].shape[0]
 
     @staticmethod
-    def _batch_generator(X1, X2, y, idxs, batch_size, num_batches):
+    def _batch_generator(X1: np.ndarray, X2: np.ndarray,
+                         y: np.ndarray, idxs: np.ndarray,
+                         batch_size: Optional[int],
+                         num_batches: int) -> Iterable[List[np.ndarray]]:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]: Should be:
```suggestion
                         num_batches: int) -> Generator[type1, type2, type3]:
```

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 8]
```diff
@@ -109,7 +116,8 @@ def _batch_generator(X1, X2, y, idxs, batch_size, num_batches):
             b = a + batch_size
             yield [X1[idxs[a:b]], X2[idxs[a:b]]], y[idxs[a:b]]
 
-    def _get_batch_generator(self, X1, X2, y, batch_size):
+    def _get_batch_generator(self, X1: np.ndarray, X2: np.ndarray,
+                             y: np.ndarray, batch_size: Optional[int]) -> Callable[..., Iterable]:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]:
```suggestion
                             y: np.ndarray, batch_size: Optional[int]) -> Callable[..., Generator[t1, t2, t3]]:
```

**Review by alexey-khorkin — CHANGES_REQUESTED** [review]

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 9]
```diff
@@ -16,8 +20,8 @@
 class NLPDataLoader(KerasDataLoader):
     """NLP Dataloader template."""
 
-    def __init__(self, collaborator_count, split_ratio,
-                 num_samples, data_path, batch_size, **kwargs):
+    def __init__(self, collaborator_count: int, split_ratio: float,
+                 num_samples: int, data_path: str, batch_size: Optional[int], **kwargs) -> None:
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]: Looks like `batch_size = None` is invalid

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py (hunk 7 above)
Comment (dmitryagapov) [reply openfl-workspace/keras_nlp/src/nlp_dataloader.py]: I think that using `Iterable` instead of `Generator` still fine.

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 10]
```diff
@@ -91,7 +95,10 @@ def get_valid_data_size(self):
         return self.X_valid[0].shape[0]
 
     @staticmethod
-    def _batch_generator(X1, X2, y, idxs, batch_size, num_batches):
+    def _batch_generator(X1: np.ndarray, X2: np.ndarray,
+                         y: np.ndarray, idxs: np.ndarray,
+                         batch_size: Optional[int],
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]: batch_size couldn't be `None`

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 11]
```diff
@@ -181,7 +189,10 @@ def get_datasets_(encoder_input_data, decoder_input_data,
     return results
 
 
-def load_shard(collaborator_count, shard_num, data_path, num_samples, split_ratio):
+def load_shard(
+    collaborator_count: int, shard_num: Any, data_path: str,
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]: Could we don't use `Any`

**Review by dmitryagapov — CHANGES_REQUESTED** [review]

File: openfl-workspace/keras_nlp/src/dataloader_utils.py (hunk 11 above)
Comment (igor-davidyuk) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]:
```suggestion
    collaborator_count: int, shard_num: int, data_path: str,
```

Comment (igor-davidyuk) [reply openfl-workspace/keras_nlp/src/dataloader_utils.py]: So we can remove `shard_num = int(shard_num)` line.
But we have to cast the value to int here https://github.com/intel/openfl/blob/fef41fb874a092e97dfa532f53fdae7bf3dcb1a5/openfl-workspace/keras_nlp/src/nlp_dataloader.py#L35

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py (hunk 10 above)
Comment (igor-davidyuk) [reply openfl-workspace/keras_nlp/src/nlp_dataloader.py]: I am also curious why is it `Optional`? Did you discover any case when we do not pass `int` here?

Comment (ViktoriiaRomanova) [reply openfl-workspace/keras_nlp/src/nlp_dataloader.py]: No, I made a mistake

**Review by igor-davidyuk — APPROVED** [review]

File: openfl-workspace/keras_nlp/src/dataloader_utils.py [hunk 12]
```diff
@@ -15,7 +17,8 @@
 logger = getLogger(__name__)
 
 
-def download_data_():
+@staticmethod
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/dataloader_utils.py]: Why is `@staticmethod` used this?

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 13]
```diff
@@ -47,11 +51,11 @@ def __init__(self, collaborator_count, split_ratio,
         self.X_valid = [valid[0], valid[1]]
         self.y_valid = valid[2]
 
-    def get_feature_shape(self):
+    def get_feature_shape(self) -> Tuple[int, ...]:
         """Get the shape of an example feature array."""
         return self.X_train[0].shape
 
-    def get_train_loader(self, batch_size=None):
+    def get_train_loader(self, batch_size: Optional[int] = None) -> Iterator[List[np.ndarray]]:
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py:59]: `batch_size` couldn't be `None`

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 14]
```diff
@@ -109,7 +116,9 @@ def _batch_generator(X1, X2, y, idxs, batch_size, num_batches):
             b = a + batch_size
             yield [X1[idxs[a:b]], X2[idxs[a:b]]], y[idxs[a:b]]
 
-    def _get_batch_generator(self, X1, X2, y, batch_size):
+    def _get_batch_generator(self, X1: np.ndarray, X2: np.ndarray,
+                             y: np.ndarray,
+                             batch_size: Optional[int]) -> Iterator[List[np.ndarray]]:
```
Comment (dmitryagapov) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py]:
```suggestion
                             batch_size: Optional[int] = None) -> Iterator[List[np.ndarray]]:
```

**Review by dmitryagapov — CHANGES_REQUESTED** [review]

File: openfl-workspace/keras_nlp/src/dataloader_utils.py (hunk 12 above)
Comment (ViktoriiaRomanova) [reply openfl-workspace/keras_nlp/src/dataloader_utils.py]: Good question, I'll fix this.

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py (hunk 13 above)
Comment (ViktoriiaRomanova) [reply openfl-workspace/keras_nlp/src/nlp_dataloader.py:59]: I don't think that we should change logic here. `batch_size = None` was put here by the developer for some purpose, moreover we compare `batch_size` with `None` at  `_get_batch_generator(..)` function

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py (hunk 14 above)
Comment (ViktoriiaRomanova) [reply openfl-workspace/keras_nlp/src/nlp_dataloader.py]: We never use `batch_size` in `self._get_batch_generator(...)` as optional argument. So I don't think we should add `Optional[int] = None`, but we can get `None` as value, that's why I guess here should be `Optional[int]`

File: openfl-workspace/keras_nlp/src/nlp_dataloader.py [hunk 15]
```diff
@@ -109,7 +117,9 @@ def _batch_generator(X1, X2, y, idxs, batch_size, num_batches):
             b = a + batch_size
             yield [X1[idxs[a:b]], X2[idxs[a:b]]], y[idxs[a:b]]
 
-    def _get_batch_generator(self, X1, X2, y, batch_size):
+    def _get_batch_generator(self, X1: np.ndarray, X2: np.ndarray,
+                             y: np.ndarray,
+                             batch_size: Union[int, None]) -> Iterator[List[np.ndarray]]:
```
Comment (alexey-khorkin) [inline openfl-workspace/keras_nlp/src/nlp_dataloader.py:122]:
```suggestion
                             batch_size: Optional[int]) -> Iterator[List[np.ndarray]]:
```

**Review by alexey-khorkin — APPROVED** [review]

**Review by dmitryagapov — APPROVED** [review]

**Review by alexey-gruzdev — APPROVED** [review]

**Pull request merged**